### PR TITLE
Fix softmax parsing in pytorch and add test

### PR DESCRIPTION
**Pull request by JanFSchulte**
Fixes issues with softmax activation in pytorch parser reported in https://github.com/fastmachinelearning/hls4ml/issues/1083 and adds softmax activation to the pytest. 

I also added a one-line change to fix https://github.com/fastmachinelearning/hls4ml/issues/1074 while I'm messing with the pytorch parser. 

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)

## Tests

Added softmax to the pytest, where it works now. 

**Test Configuration**:

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have installed and run `pre-commit` on the files I edited or added.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: hls4ml/converters/pytorch/core.py [hunk 1]
```diff
@@ -62,9 +62,13 @@ def parse_activation_layer(operation, layer_name, input_names, input_shapes, nod
             layer['activation'] = 'ThresholdedReLU'
             if layer['activ_param'] < 0:
                 raise Exception('negative threshold values not supported')
-
-        if hasattr(node, 'dim'):
+        if hasattr(class_object, 'dim'):
             layer['axis'] = class_object.dim
+            if layer['class_name'] == 'Softmax' and layer['axis'] is None:
+                layer['axis'] = -1
+            if 'IOType' in config:
```
Comment (vloncar) [inline hls4ml/converters/pytorch/core.py:69]: We have done these types of checks before, but now I think we should parse everything into the IR, and then backbends should check if the feature is supported. We can have optimizers for that. Check out feature_check.py in Vitis backend. We could add similar checks for all backends and move away from the parser (in a future PR)

**Review by jmitrevs — APPROVED** [review]
